### PR TITLE
fix signature of frame.filter

### PR DESCRIPTION
**Pull request by refack**
`items` can be `ListLike`

It is turned into and `Index` which can handle and `ListLike` as data
https://github.com/pandas-dev/pandas/blob/9b375be5aa3610e8a21ef0b5b81e4db04270f3d3/pandas/core/generic.py#L5448

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you add a test that fails without your PR and works with your PR?

**Comment by refack** [comment]
> Can you add a test

Allready started (great minds and such)

**Comment by Dr-Irv** [comment]
CI is failing because pyright got upgraded and we have to remove some `#pyright ignore` statements. This is in PR #967 .  Once that is merged in, then you can merge in the new `main` in your branch and push, and I will test the CI again.

**Comment by Dr-Irv** [comment]
> CI is failing because pyright got upgraded and we have to remove some `#pyright ignore` statements. This is in PR #967 . Once that is merged in, then you can merge in the new `main` in your branch and push, and I will test the CI again.

This is now merged, so go ahead and merge with `main` and push

File: pandas-stubs/core/frame.pyi
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:1714]: `Iterable` is too wide.   That will accept a plain string.  Can you use `ListLike` here

While the pandas code accepts a set, that's not the intent and why the docs say "list-like".  That would also mean you should remove the test that passes a set.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/core/frame.pyi
Comment (refack) [reply pandas-stubs/core/frame.pyi:1714]: I used `ListLike` in the first iteration but then mypy fails for `set` or `generator`

```console
tests\test_frame.py:323: error: Value of type variable "ListLike" of "filter" of "DataFrame" cannot be "set[str]"  [type-var]
tests\test_frame.py:325: error: Value of type variable "ListLike" of "filter" of "DataFrame" cannot be "Generator[str, None, None]"  [type-var]
```

I skimmed the code for something in the middle... Maybe `Iterable[str]`?

Comment (refack) [reply pandas-stubs/core/frame.pyi:1714]: Since `items` is converted into an `Index` it's very flexible. And `Index.__new__` has a very complicated signature with multiple overloads.
Your call as to which way to go; stricter and have false negatives, or loose and have false positives.

Comment (Dr-Irv) [reply pandas-stubs/core/frame.pyi:1714]: the rule we are using in the stubs is to base things on the docs.  So while `filter` does accept a `set`, it probably shouldn't, given the docs.  So I am for removing the test for sets, and if you want to include a generator, you can use `ListLike | Generator[Hashable, None, None]`

Comment (refack) [reply pandas-stubs/core/frame.pyi]: Done

**Review by Dr-Irv — APPROVED** [review]
Thanks @refack

**Comment by refack** [comment]
> Thanks @refack

Keep up the great work

**Comment by refack** [comment]
@Dr-Irv apropos, I just stumbled on https://github.com/pandas-dev/pandas/issues/47215
LOL
It's nice to see things stay consistent even after 2 years 👍

**Pull request merged**
